tests/suites/test_suite_cmac.data

```diff
@@ -22,15 +22,15 @@ mbedtls_cmac_setkey:MBEDTLS_CIPHER_DES_EDE3_ECB:192:0
 
 CMAC init #5 AES-224: bad key size
 depends_on:MBEDTLS_AES_C
-mbedtls_cmac_setkey:MBEDTLS_CIPHER_ID_AES:224:MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA
+mbedtls_cmac_setkey:MBEDTLS_CIPHER_AES_128_ECB:224:MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA
```

Comment:
This will cause a conflict with https://github.com/Mbed-TLS/mbedtls/pull/6385/files#diff-b55f0cc7bb0504cadf4963d58857b281f7af712a7f5587e563a234d11bb4abd3R33, but we can handle it later. Just leaving here to remember :)

Reply:
OTOH, shouldn't this be MBEDTLS_CIPHER_AES_128_CBC?

Reply:
ECB seems right. The design of CMAC is close to CBC, but it's a slightly tweaked CBC and CMAC isn't implemented on top of CBC, it's a separate implementation on top of what we call ECB but is actually just operating on one block at a time.

Reply:
Oh, OK, I just found [here ](https://github.com/Mbed-TLS/mbedtls/pull/6385/files#diff-b55f0cc7bb0504cadf4963d58857b281f7af712a7f5587e563a234d11bb4abd3R32) that there test #7 (CMAC init #7 Camellia: wrong cipher) is failing without CBC so I figured it had to be it.

Reply:
There too, CBC is not the right dependency. Neither the test code nor the implementation of CMAC refer to CBC, so I expect it's some other configuration option. `cmac.c` only supports CMAC over DES and AES, not over Camellia or ARIA, and it's not very clean about this, so maybe the problem is with the block cipher and not with the mode?

Reply:
I made this change because there was a mismatch between cipher id and cipher type (https://github.com/Mbed-TLS/mbedtls/pull/6374#issuecomment-1261867239). @AndrzejKurek do you expect some changes here?

Reply:
Another argument why ECB is correct: it's what we use for the positive cases (1 to 4 just above). And of course the negative cases should be as close as possible to the positive ones.

Reply:
OK, I can see that it's because `MBEDTLS_CIPHER_ID_CAMELLIA` (`cipher_id_t`) was used instead of a `mbedtls_cipher_type_t` type, `MBEDTLS_CIPHER_CAMELLIA_192_ECB`, and coincidentally, `MBEDTLS_CIPHER_ID_CAMELLIA` translates to 5, the same as `MBEDTLS_CIPHER_AES_128_CBC`.